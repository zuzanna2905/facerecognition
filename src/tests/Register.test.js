import {shallow} from 'enzyme';
import React from 'react';
import Register from '../component/Register/Register';

it('renders Register without crashing', () => {
    let  wrapper = shallow(<Register />);
    expect(wrapper).toMatchSnapshot();
});

it('simulate Register', () => {    
    const expectedState = {
        email: 'john@gmail.com',
        password: 'cookies',
        name: 'John'
    }
    let wrapper2 = shallow(<Register />);
    wrapper2.find('[id="name"]').simulate('change', {target: {value: 'John'}});
    wrapper2.find('[id="password"]').simulate('change', {target: {value: 'cookies'}});
    wrapper2.find('[id="email-address"]').simulate('change', {target: {value: 'john@gmail.com'}});
    expect(wrapper2.state().name).toEqual(expectedState.name);
    expect(wrapper2.state().password).toEqual(expectedState.password);
    expect(wrapper2.state().email).toEqual(expectedState.email);
    wrapper2.find('[id="registersubmit"]').simulate('click');
})