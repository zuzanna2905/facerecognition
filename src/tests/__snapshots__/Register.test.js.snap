// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Register without crashing 1`] = `
<article
  className="br3 ba dark-gray b--black-10 mv4 w-100 w-50-m w-25-l mw6 shadow-5 center"
>
  <main
    className="pa4 black-80"
  >
    <div
      className="measure"
    >
      <fieldset
        className="ba b--transparent ph0 mh0"
        id="sign_up"
      >
        <legend
          className="f2 fw6 ph0 mh0"
        >
          Register
        </legend>
        <div
          className="mt3"
        >
          <label
            className="db fw6 lh-copy f6"
            htmlFor="name"
          >
            Name
          </label>
          <input
            className="pa2 input-reset ba bg-transparent hover-bg-black hover-white w-100"
            id="name"
            name="name"
            onChange={[Function]}
            type="text"
          />
        </div>
        <div
          className="mt3"
        >
          <label
            className="db fw6 lh-copy f6"
            htmlFor="email-address"
          >
            Email
          </label>
          <input
            className="pa2 input-reset ba bg-transparent hover-bg-black hover-white w-100"
            id="email-address"
            name="email-address"
            onChange={[Function]}
            type="email"
          />
        </div>
        <div
          className="mv3"
        >
          <label
            className="db fw6 lh-copy f6"
            htmlFor="password"
          >
            Password
          </label>
          <input
            className="b pa2 input-reset ba bg-transparent hover-bg-black hover-white w-100"
            id="password"
            name="password"
            onChange={[Function]}
            type="password"
          />
        </div>
      </fieldset>
      <div
        className=""
      >
        <input
          className="b ph3 pv2 input-reset ba b--black bg-transparent grow pointer f6 dib"
          id="registersubmit"
          onClick={[Function]}
          type="submit"
          value="Register"
        />
      </div>
    </div>
  </main>
</article>
`;
